CRM/Core/Form.php

```diff
@@ -1342,14 +1342,24 @@ public function addDateRange($name, $from = '_from', $to = '_to', $label = 'From
    * @param bool $required
    * @param string $fromLabel
    * @param string $toLabel
+   * @param array $additionalOptions
+   * @param string $to string to append to the to field.
+   * @param string $from string to append to the from field.
    */
-  public function addDatePickerRange($fieldName, $label, $isDateTime = FALSE, $required = FALSE, $fromLabel = 'From', $toLabel = 'To') {
+  public function addDatePickerRange($fieldName, $label, $isDateTime = FALSE, $required = FALSE, $fromLabel = 'From', $toLabel = 'To', $additionalOptions = [],
+    $to = '_high', $from = '_low') {
```

Comment:
@seamuslee001 so my big hangup here is that I feel like we are perpetuating unnecessary complexity - for coders & users but having '_high' & '_low' in advanced search & from / to here.

I feel like w could avoid handling 'from' 'to' (but still permit the labels if we just handled 'from' & 'to' in the place where we accept incoming url-params

Reply:
@eileenmcnaughton maybe but also it would mean having to frutz with a bunch of reports that look for that when building their where clauses, i started going down that but i figure its a bit of a rabbit hole and also how would we handle it for any custom reports where as at least they are known situations in reports and we could document the odd ball ness.

This is just the core places that i can find 

```
CRM/Report/Form/Activity.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/ActivitySummary.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/ActivitySummary.php:                CRM_Utils_Array::value("activity_date_time_to", $this->_params)
CRM/Report/Form/ActivitySummary.php:            $url[] = "activity_date_time_from={$from}&activity_date_time_to={$to}";
CRM/Report/Form/Campaign/SurveyDetails.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/Case/Detail.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/Case/Detail.php:      CRM_Utils_Array::value('last_completed_date_time_to', $this->_params)
CRM/Report/Form/Case/Detail.php:      CRM_Utils_Array::value('last_activity_date_time_to', $this->_params)
CRM/Report/Form/Case/TimeSpent.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/Case/Demographics.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/Contact/Detail.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/Contact/CurrentEmployer.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/Contact/Relationship.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/Contribute/History.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/Contribute/Repeat.php:        $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/Contribute/Repeat.php:      $checkDate['receive_date1']['receive_date1_to'] = $fields['receive_date1_to'];
CRM/Report/Form/Contribute/Repeat.php:      $checkDate['receive_date2']['receive_date2_to'] = $fields['receive_date2_to'];
CRM/Report/Form/Contribute/Repeat.php:        ($error_data[$date_range . '_to']['valid'] == 'false')
CRM/Report/Form/Contribute/Repeat.php:          ($error_data[$date_range . '_to']['is_empty'] == 'true')
CRM/Report/Form/Contribute/Repeat.php:        if ($error_data[$date_range . '_to']['is_empty'] == 'false') {
CRM/Report/Form/Contribute/Repeat.php:          $errors[$date_range . '_to'] = $errorText;
CRM/Report/Form/Contribute/Repeat.php:        ($error_data[$date_range . '_to']['valid'] == 'false')
CRM/Report/Form/Contribute/Repeat.php:        if ($error_data[$date_range . '_to']['is_empty'] == 'false') {
CRM/Report/Form/Contribute/Repeat.php:          $errors[$date_range . '_to'] = $errorText;
CRM/Report/Form/Contribute/Repeat.php:        ($error_data[$date_range . '_to']['valid'] == 'true')
CRM/Report/Form/Contribute/Repeat.php:      CRM_Utils_Array::value("receive_date1_to", $this->_params)
CRM/Report/Form/Contribute/Repeat.php:      CRM_Utils_Array::value("receive_date2_to", $this->_params)
CRM/Report/Form/Contribute/Repeat.php:      $dateUrl .= "receive_date1_to={$to1}&";
CRM/Report/Form/Contribute/Repeat.php:      $dateUrl .= "receive_date2_to={$to2}&";
CRM/Report/Form/Contribute/RecurSummary.php:    $startDateFrom = CRM_Utils_Array::value("start_date_to", $this->_params);
CRM/Report/Form/Contribute/Recur.php:        $to = CRM_Utils_Array::value("calculated_end_date_to", $this->_params);
CRM/Report/Form/Contribute/Bookkeeping.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/Contribute/Summary.php:          "reset=1&force=1&receive_date_from={$dateStart}&receive_date_to={$dateEnd}",
CRM/Report/Form/Contribute/PCP.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/Contribute/TopDonor.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/Contribute/HouseholdSummary.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/Event/ParticipantListing.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/Event/IncomeCountSummary.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/Event/Summary.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/Grant/Statistics.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/Mailing/Bounce.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/Mailing/Summary.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/Member/Lapse.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/Member/Summary.php:          "reset=1&force=1&membership_join_date_from={$dateStart}&membership_join_date_to={$dateEnd}{$typeUrl}{$statusUrl}",
CRM/Report/Form/Membership/Summary.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/Pledge/Summary.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
CRM/Report/Form/Walklist/Walklist.php:            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
```

Reply:
@eileenmcnaughton if you have ideas on how we can deal with the above and whatever else is out there then I'm open to suggestions but if figured it was kind of too much work to try and get reports across the line

Reply:
@seamuslee001 yeah & I don't think I can focus on this one in time for 5.20 but reports always felt like a stretch goal. I'd rather focus on some more checks on search & seeing where the gaps are in terms of url params in the last few days before 5.20 is cut

Reply:
sure happy for this to miss 5.20 but managed to get it to convert locally so put it up anyway